### PR TITLE
feat: integration zerion ws api

### LINKED ISSUE #2578: [Demand] Integration Zerion API
## Catalogue

- [ ] UI/UX
- [x] Mask Core Feature
- [ ] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

The API documentation link: https://docs.zerion.io/zerion-api

What do those APIs can be built into Mask?

+ subscribe assets list
+ subscribe  transaction list

## Part 2: Problem to Solve

Add another Defi data source besides debank.

### PR DESCRIPTION
**Pull request by guanbinrui**
Learn more  https://docs.zerion.io/zerion-api.

close #2578

- [x]  subscribe assets list.
+ [x] subscribe transaction list.

image

### CODE REVIEW

**Comment by nuanyang233** [comment]
The proxy needs to change the Origin to `https://app.zerion.io` in order for the request to work.

image

File: packages/maskbook/src/plugins/Wallet/apis/zerion.ts [hunk 1]
```diff
@@ -1,6 +1,56 @@
-// const ZERION_API = 'wss://api-v4.zerion.io/socket.io/?api_token=Zerion.oSQAHALTonDN9HYZiYSX5k6vnm4GZNcM&EIO=3&transport=websocket'
-const ZERIO_API = 'wss://zerion-api-v4.mk.network'
+import io from 'socket.io-client'
+import type { SocketRequestBody, SocketNameSpace, SocketResponseBody, TransactionResponseBody } from '../types'
 
-export function getTransactionList() {
-    return []
+const ZERION_API = 'wss://zerion-api-v4-agent.r2d2.to/'
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/zerion.ts:4]:
```suggestion
const ZERION_API = 'wss://zerion-api-v4-agent.r2d2.to'
```

File: packages/maskbook/src/plugins/Wallet/apis/zerion.ts [hunk 2]
```diff
@@ -1,6 +1,56 @@
...
+
+//TODO: get token from ci env
+const ZERION_TOKEN = 'Demo.ukEVQp6L5vfgxcz4sBke7XvS873GMYHy'
+
+export const addressSocket = {
+    namespace: 'address',
+    socket: io(`${ZERION_API}address`, {
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/apis/zerion.ts:11]:
```suggestion
    socket: io(`${ZERION_API}/address`, {
```

**Comment by Jack-Works** [comment]
@guanbinrui Hi you should not set released as the merge target. The content will lost.

**Review by Jack-Works — APPROVED** [review]
LGTM on style

**Pull request merged**